### PR TITLE
feature/dep5 error handling

### PR DESCRIPTION
**Pull request by nicorikken**
- chore: suppress dep5 stacktraces
- feat: handle dep5 exceptions on report level

The main trick here is to use Dill to serialize the dep5 object. This serialization is necessary to cross the multiprocessing boundary.

- [x] Check Dill availability in package managers https://packages.debian.org/search?keywords=python3-dill

Availability is quite good:
- https://repology.org/project/dill/versions
- https://repology.org/project/python:dill/versions

### CODE REVIEW

**Comment by nicorikken** [comment]
Example CLI output from the current state with a malformed dep5 file:
```
$ poetry run reuse --root tmp/fake_repository/ lint
reuse.report - ERROR - .reuse/dep5 has syntax errors
# UNUSED LICENSES

The following licenses are not used:
* Autoconf-exception-3.0
* CC0-1.0
* GPL-3.0-or-later
* LicenseRef-custom


# SUMMARY

* Bad licenses:
* Deprecated licenses:
* Licenses without file extension:
* Missing licenses:
* Unused licenses: Autoconf-exception-3.0, CC0-1.0, GPL-3.0-or-later, LicenseRef-custom
* Used licenses:
* Read errors: 0
* Files with copyright information: 0 / 0
* Files with license information: 0 / 0

Unfortunately, your project is not compliant with version 3.0 of the REUSE Specification :-(
```

File: src/reuse/project.py [hunk 1]
```diff
@@ -331,11 +305,32 @@ def _licenses(self) -> Dict[str, Path]:
         return license_files
 
 
-def create_project() -> Project:
+def create_dep5(dep5_path: PathLike):
```
[the remark this answers is not in the document]
Comment (nicorikken) [reply src/reuse/project.py:308]: Good find, thanks!

**Comment by nicorikken** [comment]
In it's current state the output is even more concise:
```
$ poetry run reuse --root tmp/fake_repository/ lint
reuse._main - ERROR - .reuse/dep5 has syntax errors
```

**Comment by carmenbianca** [comment]
Calling a deserialiser on every call to `spdx_info_of` seems a little wasteful to me.

Maybe we should just get rid of python-debian already? #466

**Comment by nicorikken** [comment]
So to get this forward we have to deprecate Python 3.6, check the latest commit for the technical details.

**Comment by nicorikken** [comment]
Current behavior with logging the original error:
```
$ poetry run reuse --root tmp/dep5_syntax_fault_repo/
reuse.project - ERROR - .reuse/dep5 error: Duplicate field "Copyright" in paragraph number 0
reuse._main - ERROR - .reuse/dep5 syntax error
```

```
$ poetry run reuse --root tmp/dep5_nontext_repo/
reuse.project - ERROR - .reuse/dep5 error: 'utf-8' codec can't decode byte 0x89 in position 0: invalid start byte
reuse._main - ERROR - .reuse/dep5 could not be parsed as utf-8
```

**Comment by nicorikken** [comment]
Todo: rebase and check in a lockfile with Poetry version 1.2

**Comment by nicorikken** [comment]
Maybe rather than serializing objects using Dill, a proxy object can be defined that holds the necessary information for multiprocessing but is more specific and can be serialized. Something I should look into.

**Comment by carmenbianca** [comment]
@nicorikken There's an easier solution that I'm not sure I mentioned to you. @floriansnow worked on a microlibrary for parsing dep5 files. It's likely that this microlibrary plays nicer with serialising.

**Comment by nicorikken** [comment]
Looking deeper at pickle support for python-debian, it should be possible to serialize the underlying `data` attribute of the super class `deb822.Deb822` which is used for the specific `Header`, `FilesParagraph` and `LicenseParagraph`. Having a `copyright.Copyright` object `my_copyright` we can do: `list(map(lambda p: dict(p._RestrictedWrapper__data), my_copyright.all_paragraphs()))`

Adding pickle support for python-debian would involve fixing the weakref error by providing correct methods to sidestep the issue https://docs.python.org/3/library/pickle.html

**Comment by carmenbianca** [comment]
Fixed by #841

**Pull request closed without merging**